doc/examples/filters/plot_cycle_spinning.py

```diff
@@ -0,0 +1,91 @@
+"""
+=================
+Wavelet denoising
+=================
+
+The discrete wavelet transform is not `shift-invariant`_.  Shift variance can
+be achieved through an undecimated wavelet transform (also called stationary
+wavelet transform), at cost of increased redundancy (i.e. more wavelet
+coefficients than input image pixels).  An alternative way to approximate
+shift-invariance in the context of image denoising with the discrete wavelet
+transform is to use the technique known as "cycle spinning".  Essentially this
+involves averaging the results of the following 3-step procedure for multiple
+spatial shifts, n:
+
+1.) (circularly) shift the signal by an amount, n
+2.) apply denoising
+3.) apply the inverse shift
+
+For 2D image denoising, we demonstrate here that such cycle-spinning can
+provide a substantial increase in quality, with much of the gain being
+achieved simply by averaging shifts of only n=0 and n=1 on each axis.
+
+.. _`shift-invariant`: https://en.wikipedia.org/wiki/Shift-invariant_system
+"""
+import matplotlib.pyplot as plt
+
+from skimage.restoration import denoise_wavelet, cycle_spin
+from skimage import data, img_as_float
+from skimage.util import random_noise
+from skimage.measure import compare_psnr
+
+
+original = img_as_float(data.chelsea()[100:250, 50:300])
+
+sigma = 0.155
+noisy = random_noise(original, var=sigma**2)
+
+fig, ax = plt.subplots(nrows=2, ncols=3, figsize=(10, 4), sharex=False,
+                       sharey=False, subplot_kw={'adjustable': 'box-forced'})
+ax = ax.ravel()
+plt.gray()
```

Comment:
I'm not a fan of this construct... (as opposed to explicitly using `cmap='gray'`.) And at any rate, it seems totally useless in this context?

Fantastic example otherwise though! =)

Reply:
In other news, I think you can totally bump minimum mpl to 1.5.

Reply:
That `plt.gray()` wasn't even doing anything as the images are RGB.  not sure where it came from!